storage/innobase/handler/ha_innodb.cc

```diff
@@ -7798,12 +7798,32 @@ int ha_innobase::intrinsic_table_write_row(uchar *record) {
       convert_error_code_to_mysql(err, m_prebuilt->table->flags, m_user_thd));
 }
 
+static ulint get_temp_tablespace_size(const ibt::Tablespace *ts) noexcept {
+  if (ts == nullptr) {
+    return 0;
+  }
+  fil_space_t *space = fil_space_get(ts->space_id());
+  if (space == nullptr) {
+    return 0;
+  }
+  const page_size_t page_size(space->flags);
+  return space->size * page_size.physical();
```

Comment:
Do you know if innodb temporary tablespaces shrink after a query finishes execution? If it doesn't, then we could be miscounting. It would be nice to have a test case that proves this.

Reply:
We are miscounting, as anything that doesn't increment the file size (e.g. many updates, delete/drop, etc) isn't counted.

Otherwise, I don't think the file size decreases during the lifetime of a session, and that's why this approximation should work.

Reply:
With some testing, it looks like the tablesize gets reset when the session is closed and the temp table is deleted. But while the table/connection is active, it seems to behave like other tablespace, so the space is not reclaimed from the file on deletion, nor overwrites.